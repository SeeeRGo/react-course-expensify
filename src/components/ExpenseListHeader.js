import React from 'react'
import {connect} from 'react-redux'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses.js'
import selectExpensesTotal from '../selectors/expenses-total.js'

export const ExpenseListHeader = (props) => (
  <div>
  {
    props.expenses.length !== 0 &&
    <p>
      {`You are viewing ${props.expenses.length} ${props.expenses.length === 1 ? 'expense' : 'expenses'}\n`}
      {`for total of ${numeral(selectExpensesTotal(props.expenses)).format('$0,0.00')}`}
    </p>
  }
	</div>
)

const mapStateToProps = (state) => {
	return {
    expenses: selectExpenses(state.expenses, state.filters)
	}
}
export default  connect(mapStateToProps)(ExpenseListHeader)