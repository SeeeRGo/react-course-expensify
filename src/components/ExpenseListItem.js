import React from 'react'
import {connect} from 'react-redux'
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

export const ExpenseListItem = (props) => (
	<div>
		<Link to={`/edit/${props.id}`}>
			<p>{props.description}</p>
		</Link>
		<p>
			{numeral(props.amount / 100).format('$0,0.00')}
			-
			{moment(props.createdAt).format('MMMM Do, YYYY')}
		</p>
	</div>
)

export default connect()(ExpenseListItem)