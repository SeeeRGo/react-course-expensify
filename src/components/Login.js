import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from '../actions/auth'

export class Login extends React.Component {
  render() {
    return (
      <div className="box-layout">
        <div className="box-layout__box">
        <h1 className="box-layout__title">Expensify</h1>
        <p>It's time to get your expenses under control</p>
        <button className="button-main" onClick={this.props.startLogin}>Login With Google</button>
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
})

export default connect(null, mapDispatchToProps)(Login)