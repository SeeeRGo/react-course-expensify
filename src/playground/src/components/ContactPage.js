import React from 'react'

const ContactPage = () => (
	<div>
		<h1>My contacts</h1>
		<p>Email me email@email.com</p>
	</div>
)

export default ContactPage