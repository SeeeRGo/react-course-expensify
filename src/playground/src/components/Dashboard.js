import React from 'react'

const Dashboard = () => (
	<div>
		<h1>Site with my portfolio</h1>
	</div>
)

export default Dashboard