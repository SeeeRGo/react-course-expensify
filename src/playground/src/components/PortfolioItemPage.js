import React from 'react'

const PortfolioItemPage = (props) => (
		<div>
			This is a portfolio item with id {props.match.params.id}
		</div>
	)

export default PortfolioItemPage