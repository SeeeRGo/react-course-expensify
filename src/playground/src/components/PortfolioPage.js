import React from 'react'
import {Link} from 'react-router-dom'

const PortfolioPage = () => {
	const portfolioItems = [1,3,5]
	return (
		<div>
			<h1>Portfolio Page</h1>
			{ portfolioItems.map((item, id) => {
				return (<Link key={id} to={`/portfolio/${id}`}>Item {id}</Link>)
			})}
		</div>
	)
}

export default PortfolioPage