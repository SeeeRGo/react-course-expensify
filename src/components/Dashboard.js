import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListHeader from './ExpenseListHeader'
import ExpenseListFilters from './ExpenseListFilters'

const Dashboard = () => (
	<div>
		<ExpenseListHeader />
		<ExpenseListFilters/>
		<ExpenseList/>
	</div>
)

export default Dashboard