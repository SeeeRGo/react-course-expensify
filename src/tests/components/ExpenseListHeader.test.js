import React from 'react'
import selectExpensesTotal from '../../selectors/expenses-total'
import expenses from '../fixtures/expenses'
import { shallow } from 'enzyme';
import { ExpenseListHeader } from '../../components/ExpenseListHeader';

test('should render ExpenseListHeader correctly', () => {
  const wrapper = shallow(<ExpenseListHeader expenses={expenses}/>)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseListHeader correctly', () => {
  const wrapper = shallow(<ExpenseListHeader expenses={[expenses[0]]}/>)
  expect(wrapper).toMatchSnapshot()
})