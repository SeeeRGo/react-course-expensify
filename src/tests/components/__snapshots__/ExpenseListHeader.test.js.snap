// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListHeader correctly 1`] = `
<div>
  <p>
    You are viewing 3 expenses
    
    for total of $114,195.00
  </p>
</div>
`;

exports[`should render ExpenseListHeader correctly 2`] = `
<div>
  <p>
    You are viewing 1 expense
    
    for total of $195.00
  </p>
</div>
`;
